qa-tests-backend/src/test/groovy/ProcessesListeningOnPortsTest.groovy

```diff
@@ -226,6 +238,104 @@ class ProcessesListeningOnPortsTest extends BaseSpecification {
 
         def list3 = processesListeningOnPorts.listeningEndpointsList
         assert list3.size() == 0
+
+        destroyDeployments()
+    }
+
+    @Tag("BAT")
+    def "Verify networking endpoints disappear when process is terminated"() {
+        given:
+        "When a deployment listening on a port is created and then the process is terminated"
+
+        setupSpec()
+        rebuildForRetries()
+        def clusterId = ClusterService.getClusterId()
+
+        String deploymentId3 = targetDeployments[2].getDeploymentUid()
+
+        def gotCorrectNumElements = waitForResponseToHaveNumElements(1, deploymentId3, 240)
+
+        // First check that the listening endpoint appears in the API
+        assert gotCorrectNumElements
+
+        def processesListeningOnPorts = evaluateWithRetry(10, 10) {
```

Comment:
I wonder why didn't you return the `getProcessesListeningOnPortsResponse` response on the `waitForResponseToHaveNumElements` call? 

The test is having to make the request twice. First when running `waitForResponseToHaveNumElements` and then again here to make the content assertions. 

Wouldn't it be simpler to:

```groovy
def processListeningOnPorts = waitForResponseToHaveNumElements(1, deploymentId3, 240)

// make assertions
assert processesListeningOnPorts
def list = processesListeningOnPorts.listeningEndpointsList
assert list.size() == 1
```
You're even making the assertion for the size twice (one inside of `waitForResponseToHaveNumElements` and another here ☝️ )

Reply:
I have changed this now to what you suggested. I knew it was better to return processesListeningOnPorts than a boolean, but I am not experienced with groovy and I had some difficulty doing that. I would need to declare processesListeningOnPorts outside of the loop and specify its type. I had some difficulty doing that.